### PR TITLE
feat(mixer): added Mixer provider

### PR DESCRIPTION
**Pull request by MattBroach**
# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must be 100% pep8 and isort clean.
 - [x] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [x] Feel free to add yourself to `AUTHORS`.
 
 ## Provider Specifics
 
 In case you add a new provider:
 
- [x] Make sure unit tests are available.
- [x] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [x] Add documentation to `docs/providers.rst`.
- [x] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

File: allauth/socialaccount/providers/mixer/views.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class MixerOAuth2Adapter(OAuth2Adapter):
+    provider_id = MixerProvider.id
+    access_token_url = 'https://mixer.com/api/v1/oauth/token'
+    authorize_url = 'https://mixer.com/oauth/authorize'
+    profile_url = 'https://mixer.com/api/v1/users/current'
+
+    def complete_login(self, request, app, token, **kwargs):
+        headers = {'Authorization': 'Bearer {}'.format(token.token)}
+        response = requests.get(self.profile_url, headers=headers)
+
```
Comment (pennersr) [inline allauth/socialaccount/providers/mixer/views.py]: Can you add a response.raise_for_status() here?

File: tox.ini [hunk 2]
```diff
@@ -3,7 +3,7 @@ envlist =
     py{35,36,37}-django20
     py{35,36,37}-django21
     py{35,36,37}-django22
-    py{35,36,37}-django30
+    py{36,37,38}-django30
```
Comment (pennersr) [inline tox.ini]: This change is not related to Mixer, can you please file a separate PR if changes are needed here?

Comment (MattBroach) [reply tox.ini]: I've revered this change in this PR.  At least locally, this was causing tox to fail (since python 3.5 isn't in the supported python versions for Django 3.0), but I have no idea if this will affect the travis build or not.

**Comment by pennersr** [comment]
Merged via 278c0ae8
